OlinkAnalyze/vignettes/bridging_introduction.Rmd

```diff
@@ -0,0 +1,233 @@
+-   plate control normalized only and run conditions (e.g lab and
+    reagent lots) have changes.
+
+-   intensity normalized but from two different sample populations.
+
+To bridge two or more Olink® NPX datasets, bridging samples are needed
+to calculate adjustment factors between datasets. Bridging samples are 
+overlapping samples between datasets. The recommended number
+of bridging samples for Explore 1536 datasets is between 8-16. Olink® 
+NPX datasets without overlapping samples can not be combined to perform 
+joint analysis using the bridging approach described below.
+
+The following tutorial is designed to give you an overview of the kinds
+of data combining methods that are possible using the Olink® bridging
+procedure.
+
+## Setup bridging datasets
+
+```{r setup}
+library(OlinkAnalyze)
+library(dplyr)
+library(stringr)
+library(ggplot2)
+library(kableExtra)
+```
+
+Bridging datasets are standard Olink® NPX tables. They can be loaded using 
+`read_NPX()` function with NPX manager output file as input.
+
+```{r message=FALSE, eval=FALSE, echo = TRUE}
+data1 <- read_NPX("~/NPX_file1_location.xlsx")
+data2 <- read_NPX("~/NPX_file2_location.xlsx")
+```  
+
+To demonstrate how bridging works, we will use the example datasets
+(`npx_data1` and `npx_data2`) from **OlinkAnalyze** package.
+
+## Check bridging datasets
+
+Explore and have an overview of the datasets that are going to be bridged. For 
+example, plot and compare NPX distribution between datasets.
+
+```{r check, message=FALSE}
+# Load datasets
+npx_1 <- npx_data1 %>% 
+  filter(!str_detect(SampleID, 'CONTROL_SAMPLE')) %>% #Remove control samples
+  mutate(dataset = "data1")
+npx_2 <- npx_data2 %>% 
+  filter(!str_detect(SampleID, 'CONTROL_SAMPLE')) %>% #Remove control samples
+  mutate(dataset = "data2")
+
+npx_df <- bind_rows(npx_1,npx_2)
+
+
+# Plot NPX density before bridging normalization
+npx_df %>% 
+  mutate(Panel = gsub("Olink ", "", Panel)) %>% 
+  ggplot(aes(x = NPX, fill = Panel)) +
+  geom_density(scale = 1, alpha = 0.7) + 
+  facet_grid(dataset~Panel) +
+  olink_fill_discrete(coloroption = c('teal', 'lightblue')) +
+  set_plot_theme() + 
+  ggtitle("Before bridging normalization: NPX distribution") +
+  theme(axis.title.x = element_blank(),
+        axis.title.y = element_blank(),
+        strip.text = element_text(size=16),
+        legend.title = element_blank(),
+        legend.position = "none")
+```  
+
+Use PCA plot to visualize sample-to-sample distance before bridging.
+
+```{r pca1, message=FALSE, fig.cap="PCA plot of combined datasets before bridging"}
+## before bridging
+
+### Generate unique SampleIDs
+npx_1 <- npx_data1 %>% 
+  mutate(dataset = "data1") %>% 
+  mutate(SampleID = paste0(dataset, PlateID, SampleID))
+  
+npx_2 <- npx_data2 %>% 
+  mutate(dataset = "data2") %>% 
+  mutate(SampleID = paste0(dataset, PlateID, SampleID))
+             
+npx_before_br <- rbind(npx_1, npx_2)
+
+### PCA plot
+OlinkAnalyze::olink_pca_plot(df = npx_before_br,
```

Comment:
Now I have highlighted the bridging samples in PCA. Please let know if that is what you have in mind. I do not really understand what you meant by "potentially exclude them as outliers", could you please elaborate on that?

Reply:
I was thinking that if bridge samples are found to be outliers in the PCA plot, they should not be used as bridge samples. So that is the reason I wanted them highlighted in the plot.